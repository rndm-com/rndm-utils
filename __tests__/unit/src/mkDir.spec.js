const fs = require('fs');
const mkDir = require('../../../src/mkDir');
const readDir = require('../../../src/readDir');
const { describe, test } = global;

describe('src/mkDir', () => {
  test('current', () => {
    const base = __dirname;
    const path = 'a/b/c';

    const before = readDir(base).find(i => i.includes('a/b/c'));
    expect(before).toBe(undefined);

    mkDir(path, base);

    const after = readDir(base).filter(i => i.includes('a/b/c')).map(i => i.replace(base, ''));
    expect(after).toMatchSnapshot();

    const folder = [base, path].join('/');

    const comps = path.split('/');

    fs.rmdirSync([base, ...comps].join('/'));
    fs.rmdirSync([base, ...comps.slice(0, comps.length - 1)].join('/'));
    fs.rmdirSync([base, ...comps.slice(0, comps.length - 2)].join('/'));

  });

  test('no path', () => {
    const base = __dirname;
    const path = undefined;

    mkDir(path, base);

    const sut = readDir(base).find(i => i.includes('a/b/c'));
    expect(sut).toBe(undefined);

  });

  test('existing', () => {
    const base = __dirname;

    fs.mkdirSync([__dirname, 'a'].join('/'));

    const path = 'a/b/c';

    const before = readDir(base).find(i => i.includes('a/b/c'));
    expect(before).toBe(undefined);

    mkDir(path, base);

    const after = readDir(base).filter(i => i.includes('a/b/c')).map(i => i.replace(base, ''));
    expect(after).toMatchSnapshot();

    const comps = path.split('/');

    fs.rmdirSync([base, ...comps].join('/'));
    fs.rmdirSync([base, ...comps.slice(0, comps.length - 1)].join('/'));
    fs.rmdirSync([base, ...comps.slice(0, comps.length - 2)].join('/'));

  });
});
