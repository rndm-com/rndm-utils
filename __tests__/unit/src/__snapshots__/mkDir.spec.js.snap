// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/mkDir current 1`] = `Array []`;

exports[`src/mkDir existing 1`] = `Array []`;
